@here/harp-mapview-decoder/lib/TilerService.ts

```diff
@@ -81,14 +81,17 @@ export class TilerService extends WorkerService {
     ): Promise<WorkerServiceResponse> {
         let tileIndex = this.m_tileIndexMap.get(message.id);
         if (tileIndex === undefined) {
-            const response = await fetch(message.url);
-            if (!response.ok) {
-                logger.error(`${message.url} Status Text:  ${response.statusText}`);
-                return { response: {} };
+            let rawJSON = message.input;
+            if (typeof message.input === "string") {
+                const response = await fetch(message.input);
+                if (!response.ok) {
+                    logger.error(`${message.input} Status Text:  ${response.statusText}`);
+                    return { response: {} };
+                }
+                const answer = await response.text();
```

Comment:
I would use `response.json()`.

This enables browser to parse json immediately while receiving without need to buffer whole string.
May be more efficient on platforms that want to leverage this opportunity.